analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/InvalidCastToInterface.CSharp10.cs

```diff
@@ -0,0 +1,15 @@
+﻿public interface IMyInterface { }
+
+public class SomeImpl : IMyInterface { }
+
+public class MyClass { }
+
+public struct S
+{
+    public void M()
+    {
+        var myclass = new MyClass();
+        MyClass a;
+        (a, var b) = (myclass, (IMyInterface)myclass); // Noncompliant, this part is not based on SE
```

Comment:
What if we add a notation so we fast see which issues are supposed to be detected by SE and which by Syntax Based rule.

Something simple like: 
For syntax based we don't add something extra
```csharp
(a, var b) = (myclass, (IMyInterface)myclass); // Noncompliant
```
and if it's SE, we can just add the chars SE at the end:

```csharp
i1 ??= (int)i1; // FIXME Non-compliant {{Nullable is known to be empty, this cast throws an exception.}} SE
```
or 
```csharp
i1 ??= (int)i1; // FIXME Non-compliant ,  SE
```

Reply:
I'll keep the syntax one and I'll add the SE not everywhere in a later PR